presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -112,4 +123,51 @@ public static boolean isEqualComparisonExpression(Expression expression)
     {
         return expression instanceof ComparisonExpression && ((ComparisonExpression) expression).getOperator() == ComparisonExpression.Operator.EQUAL;
     }
+
+    static Optional<EnumType> tryResolveEnumLiteralType(QualifiedName qualifiedName, TypeManager typeManager)
+    {
+        int nParts = qualifiedName.getParts().size();
+        if (nParts < 2) {
+            // an enum literal should be of the form `MyEnum.my_key`
+            return Optional.empty();
+        }
+        try {
+            Type baseType = typeManager.getType(parseTypeSignature(qualifiedName.getParts().get(nParts - 2)));
```

Comment:
Sorry, my previous recommendation was wrong. You actually should just use `parseTypeSignature(qualifiedName.getPrefix())` here. If the enum literal is of the form `catalog.schema.EnumType.EnumValue`, then the type should be `catalog.schema.EnumType`, rather than just `EnumType` alone.